junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/discovery/DiscoverySelectorResolver.java

```diff
@@ -64,7 +64,7 @@ public void resolveSelectors(EngineDiscoveryRequest request, TestDescriptor engi
 		request.getSelectorsByType(UniqueIdSelector.class).forEach(selector -> {
 			javaElementsResolver.resolveUniqueId(selector.getUniqueId());
 		});
-		pruneTree(engineDescriptor);
+		engineDescriptor.pruneTree();
```

Comment:
Wouldn't _If you you can do something today, why leave it  for tomorrow_ apply to this?

If DiscoverySelectorResolver can positively determine entries to be pruned why not do it?